sdk/tables/Azure.Data.Tables/src/TablesRequestFailedDetailsParser.cs

```diff
@@ -15,7 +15,7 @@ public override bool TryParse(Response response, out ResponseError error, out ID
         {
             error = null;
             data = null;
-            if (response.ContentStream == null || !response.ContentStream.CanSeek)
+            if (response.ContentStream == null || !(response.ContentStream is MemoryStream))
```

Comment:
This may need to change if we end up wrapping buffered content in a new stream type as part of the Client Model work, no?

Reply:
We use MemoryStream there also, IIRC.

Reply:
For now, but that may change as part of the "we're disposing the content stream before returning it" fix.   One of the proposed solutions is to wrap buffered content in an internal stream type to avoid disposing `MemoryStream` instances that don't belong to us.

Reply:
Yes, it would be great if we could avoid relying on this internal implementation detail as a contract, which could break since it's not part of any public API.

Reply:
Discussed a bit offline with @annelo-msft - we'll look into how this could be standardized in our internal APIs going forward.

Reply:
In fact - out of curiosity, is there a need to use ContentStream here instead of response.Content directly?  It would be great if we could just lean on the buffering Azure.Core does automatically.

Reply:
We do use `Content` to actually read.

Reply:
I see, so this line is essentially doing a TryGet over the Content field to ensure it's buffered, before accessing a property that could potentially throw?

@KrzysztofCwalina, I think this supports your argument that this property shouldn't throw anymore ... do you agree with that interpretation?

Reply:
tracking with https://github.com/Azure/azure-sdk-for-net/issues/41647